ocs_ci/ocs/jenkins.py

```diff
@@ -288,22 +291,65 @@ def create_ocs_jenkins_template(self):
         ocs_jenkins_template_obj = OCS(**tmp_dict)
         ocs_jenkins_template_obj.create()
 
-    def get_builds_logs(self):
+    def get_build_duration_time(self, namespace, build_name):
+        """
+        get build duration time
+
+        Args:
+            namespace (str): get build in namespace
+            build_name (str): the name of the jenkins build
+        """
+        ocp_obj = OCP(namespace=namespace, kind='build')
+        output_build = ocp_obj.describe(resource_name=build_name)
+        list_build = output_build.split()
+        index = list_build.index('Duration:')
+        self.build_completed[(build_name, namespace)] = list_build[index + 1]
+
+    def print_completed_builds_results(self):
         """
 
         Get builds logs and print them on table
         """
         log.info('Print builds results')
         build_table = PrettyTable()
         build_table.field_names = ["Project", "Build", "Duration"]
-        for build in self.build_completed:
-            ocp_obj = OCP(namespace=build[1], kind='build')
-            output_build = ocp_obj.describe(resource_name=build[0])
-            list_build = output_build.split()
-            index = list_build.index('Duration:')
-            build_table.add_row([build[1], build[0], list_build[index + 1]])
+        for build, time_build in self.build_completed.items():
+            build_table.add_row([build[1], build[0], time_build])
         log.info(f'\n{build_table}\n')
 
+    def get_nodes(self, node_type='worker', num_of_nodes=1):
+        """
+        get nodes
+
+        Args:
+            node_type (str): The node type (e.g. worker, master)
+            num_of_nodes (int): The number of nodes to be returned
+
+        Returns:
+            list: List of compute node names
+        """
+        if node_type == 'master':
+            nodes1 = []
+            nodes = get_typed_nodes(node_type=node_type, num_of_nodes=num_of_nodes)
+            for node in nodes:
+                nodes1.append(node.name)
+        elif node_type == 'worker':
+            pod_objs = []
+            for project in self.projects:
+                pod_names = get_pod_name_by_pattern(
+                    pattern='jenkins', namespace=project
```

Comment:
Does all the pods created in the test  has the same pattern 'jenkins' ?
(app pod, build pod etc)

Reply:
yes